test/api/buffer/receivers/delete/001-create-ok/request.json

```diff
@@ -1,9 +1,11 @@
 {
   "path": "/v1/receivers",
   "method": "POST",
+  "headers": {
+    "X-StorageApi-Token": "%%TEST_KBC_STORAGE_API_TOKEN%%"
```

Comment:
Nedame tam rovno vsade aj `Content-Type`?

Reply:
Jo ok, myslel jsem to jako BC default když headers nejsou v requestu definovaný, ale máš pravdu že je lepší když to bude povinny.

Reply:
myslim iba upravu testov, ... ak chyba content type, tak je to tusim default JSON? (alebo to spadne?, nepamatam si, moze byt test aj na to)

----

teda:
- posleme iba headers definovane v teste
- ak bude definovana header `Content-Type: ...json`, tak body zakodujeme ako json
- inak musi byt body string

Reply:
Když oddělám to explicitní nastavení content-type, tak to funguje ok.

> ak bude definovana header Content-Type: ...json, tak body zakodujeme ako json
> inak musi byt body string

Nevím jestli rozumím. Když v `request.json` bude v `body` objekt co se dekóduje do mapy, ale v `headers` nebude jsonový `content-type`, tak mám z toho body udělat string?

Reply:
mas vyhodit chybu, ze to nevies poslat, ... alebo mas iny napad?

Reply:
(nie len ked tam je JSON objekt, ale hocico ine ako `string`)

Reply:
No jestli to dobře chápu, tak to budu muset nějak takhle?
![CleanShot 2022-11-25 at 12 54 15](https://user-images.githubusercontent.com/215660/203980687-5a06521b-0d13-4c53-80f3-8d5e70958ef1.jpg)

Reply:
```
if v, ok body.(string); ok {r.SetBody(v)}
else if v, ok body.(map[string]any); ok && ct == "json" {r.SetBody(json.MustEncode(v))}
else
FailNow("body must be string, or a JSON object,  if the Content-Type is JSON)

```

Reply:
Kua to přecastovávání je divný, jakto že to teď jde bez switche? 😅 No to je jedno, spíš mi není jasný proč tam dáváš ten `json.MustEncode(v)`, Resty si s tím poradí když tam jde teď přímo ta mapa z `request.Body` (https://github.com/keboola/keboola-as-code/blob/de7733a7f315fd75385e4fc6eccecf1d95a0161b/test/api/buffer/api_test.go#L324). (To je asi to co mi není celou dobu jasný jak jsi psal že to máme "zakódovávat do jsonu" a "jinak musí být body string", protože Resty si zjevně poradí samo s čímkoli podle content type).

Reply:
kludne pouzi switch, ... su to rovnocenne moznosti

>  Resty si s tím poradí když tam jde teď přímo ta mapa z request.Body
Ok, nestudoval som to (len mi prislo, ze to riesis nejak zlozito, tak som hodil nejaky fragment kodu, ako by to mohlo asi vyzerat)

Takze je este nejaka otazka? :slightly_smiling_face:  ci uz je to jasne

Reply:
Ne, už dobrý. 🙂 👍